### ISSUE TITLE
How should tab navigation work after selecting a different screen in the navbar?

### ISSUE THREAD

**Issue by samreid** [phetsims/joist#566]
While working on https://github.com/phetsims/wave-interference/issues/412 I noticed the following pattern.  Assume that Screen 1 has the following tab traversal order, for, say 8 focusable items in Screen 1 and 9 things in Screen 2:
Screen 1: 1, 2, 3, 4, 5, 6, 7, 8, navbar.home, navbar.1, navbar.2, keyboard button, phet button.
Screen 2: 1, 2, 3, 4, 5, 6, 7, 8, 9, navbar.home, navbar.1, navbar.2, keyboard button, phet button

When the user is on Screen 1 and presses "navbar.2", the keyboard focus is still on "navbar.2".  However, the design team has set the focus order for screen to to be 1, 2, 3, 4, 5, 6, 7, 8, 9, meaning we want the user to focus on "1" first.  How is the user supposed to get to "1"?  Pressing SHIFT-TAB takes quite a lot of presses.  Pressing TAB (forward) goes past the browser address bar, and I'm not sure that intuitive.

**Comment by jessegreenberg** [comment]
My understanding is that keeping focus on "navbar.2" is the most typical behavior for a web application. It would be expected to go through the address bar when moving forward to the top of the page. It would be less intuitive to have focus removed or moved somewhere else. A11y guidelines typically warn against moving focus unnecessarily because it can be disorienting for the user.

Moving focus on activation of a button is acceptable according to WCAG so we could do something different here but I personally don't think it is necessary.

Also including @terracoda since she was involved a11y for navigation bar.

**Comment by zepumph** [comment]
I agree with @jessegreenberg, though understand why this issue was created. I also defer to @terracoda's thoughts on the matter.

**Comment by samreid** [comment]
I agree automatically transferring focus seems disorienting and I'm not recommending that.  What about a strategy like this: when you press "Tab" after changing screens in the navigation bar, the next focused item is (a) the first item in the screen instead of (b) the next item in the navigation bar?

**Comment by terracoda** [comment]
@samreid, if we treat the list of screens like a tablist, which I think is exactly the kind of interaction widget that it is. The screen buttons activate a screen just like a button in a tablist would activate a  tab panel.

Once a screen is active pressing the tab key should go to the first focus-able thing in the screen (i.e., tab panel). The arrow keys would be used to go between the list of screen buttons (the list of tab buttons).

If focus returns to the Tab list after exploring a screen focus should start on the button of the activated screen. Then arrow keys can move between screen buttons and space/enter keys would activate the screen.

There is no need to transfer focus.

I think the tricky bit for us is our tab panel is above the buttons, so when a user tabs away from the screen button to explore the active screen, should focus start at the top or at the bottom. I would think that focus should start on the most pedagogically relevant interactive item just as if a user were starting their exploration from the top.

Does this make sense? AND
Would it be possible to start focus on the top if the screen buttons are actually below the Play Area?

**Comment by terracoda** [comment]
@samreid, @jessegreenberg, and @zepumph for examples see
https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html

**Comment by terracoda** [comment]
And we do "manual activation" on the screen buttons - not "auto activation".

**Comment by jessegreenberg** [comment]
@samreid @terracoda that is an interesting idea, I like that better than moving focus. However, I still think it runs the risk of being disorienting and it feels like we are trying to predict what the user intends to do. For instance, what if the user does want to move to the next screen button and click that one just to see what is there before going through an active screen?

Would the order change be only for "tab" or for other methods of screen reader traversal too?

**Comment by jessegreenberg** [comment]
We discussed this during a meeting on 7/12/19:

Design for the navigation bar was previously explored in https://github.com/phetsims/a11y-research/issues/45. That was implemented and discussed further in https://github.com/phetsims/joist/issues/462.

@terracoda agreed with @samreid that it would be ideal if the next focusable item in the sim was the first focusable item in the screen after activating a screen button. @terracoda also thought that the navigation bar button design should be reconsidered and implemented as a "Tab Panel" where screen buttons are navigated with arrow keys (like https://dequeuniversity.com/library/aria/tabpanels-accordions/tabpanel2).

However, we are not able to explore this now but and will come back to this in Q4 of this year. Our thought for wave-interference was to publish with keyboard navigation as is. @samreid does that sound reasonable to you?
